core/src/main/scala/neotypes/Stream.scala

```diff
@@ -1,14 +1,40 @@
 package neotypes
 
+import org.reactivestreams.Publisher
+
+import scala.collection.compat.Factory
+
 @annotation.implicitNotFound("The stream type ${S} is not supported by neotypes")
 trait Stream[S[_]] {
   type F[T]
 
-  def init[T](value: () => F[Option[T]]): S[T]
+  // Legacy module ------------------------------------------------------------
+  private[neotypes] def init[A](value: () => F[Option[A]]): S[A]
+
+  private[neotypes] def onComplete[A](s: S[A])(f: => F[Unit]): S[A]
+
+  private[neotypes] def fToS[A](f: F[S[A]]): S[A]
+  // --------------------------------------------------------------------------
+
+  // New (Rx) module ----------------------------------------------------------
+  private[neotypes] def fromRx[A](publisher: Publisher[A]): S[A]
```

Comment:
Should Publisher[A] be wrapped in effect type F?

Reply:
I once had the same doubt. But, a `Publisher` is a lazy _(and somewhat pure)_ value.
One only has to ensure that it was not already subscribed by someone else.

BTW, if you look how to convert a `Publisher` into any of our streams you will see that neither requires it being wrapped in `F`.
Take a look to [this](https://github.com/BalmungSan/scala-mongodb-streaming).